pkg/client/chunk/expected_test.go

```diff
@@ -333,6 +335,53 @@ func TestRecvExpected_Timestamp_Restart(t *testing.T) {
 	assert.Empty(t, diff)
 }
 
+func TestRecvExpected_RepeatPadding(t *testing.T) {
+	const grpcMaxSize = 1024 * 1024 * 4 // GRPC default max message size
+
+	// build the units to ensure that there is more than double the units required for the GRPC configuration
+	minimumMsgSize := grpcMaxSize * 2 // double it
+	var units []*proto.UnitExpected
+	var unitsSize int
+	var nextUnitID int
+	for {
+		unit := &proto.UnitExpected{
+			Id:             fmt.Sprintf("fake-input-%d", nextUnitID),
+			Type:           proto.UnitType_INPUT,
+			State:          proto.State_HEALTHY,
+			ConfigStateIdx: 1,
+			Config: &proto.UnitExpectedConfig{
+				Id:   "testing",
+				Type: "testing",
+				Name: "testing",
+			},
+			LogLevel: proto.UnitLogLevel_ERROR,
+		}
+		units = append(units, unit)
+		unitsSize += gproto.Size(unit)
+		nextUnitID++
+
+		if unitsSize > minimumMsgSize {
```

Comment:
[Suggestion]
you could have made it part of the for condition

```go
for ; unitsSize > minimumMsgSize ; {
```

Reply:
fixed